pkg/bdba/ctl_bdba.go

```diff
@@ -0,0 +1,514 @@
+	EmailSMTPUser     string `json:"emailSMTPUser"`
+	EmailSMTPPassword string `json:"emailSMTPPassword"`
+	EmailFrom         string `json:"emailFrom"`
+	EmailSecurity     string `json:"emailSecurity"`
+	EmailVerify       bool   `json:"emailVerify"`
+
+	// LDAP Authentication
+	LDAPEnabled              bool   `json:"ldapEnabled"`
+	LDAPServerURI            string `json:"ldapServerURI"`
+	LDAPUserDNTemplate       string `json:"ldapUserDNTemplate"`
+	LDAPBindAsAuthenticating bool   `json:"ldapBindAsAuthenticating"`
+	LDAPBindDN               string `json:"ldapBindDN"`
+	LDAPBindPassword         string `json:"ldapBindPassword"`
+	LDAPStartTLS             bool   `json:"ldapStartTLS"`
+	LDAPVerify               bool   `json:"ldapVerify"`
+	LDAPRootCASecret         string `json:"ldapRootCASecret"`
+	LDAPRequireGroup         string `json:"ldapRequireGroup"`
+	LDAPUserSearch           string `json:"ldapUserSearch"`
+	LDAPUserSearchScope      string `json:"ldapUserSearchScope"`
+	LDAPGroupSearch          string `json:"ldapGroupSearch"`
+	LDAPGroupSearchScope     string `json:"ldapGroupSearchScope"`
+	LDAPNestedSearch         bool   `json:"ldapNestedSearch"`
+
+	// Logging
+	DisableFrontendLogging bool `json:"disableFrontendLogging"`
+	DisableWorkerLogging   bool `json:"disableWorkerLogging"`
+
+	// Worker scaling
+	WorkerReplicas    int `json:"workerReplicas"`
+	WorkerConcurrency int `json:"workerConcurrency"` // TODO: Patcher
+
+	// Networking and security
+	RootCASecret string `json:"rootCASecret"`
+	HTTPProxy    string `json:"httpProxy"`
+	HTTPNoProxy  string `json:"httpNoProxy"`
+
+	// Minio
+	MinioMode  string `json:"minioMode"`
+
+	// Ingress
+	IngressEnabled       bool   `json:"ingressEnabled"`
+	IngressHost          string `json:"ingressHost"`
+	IngressTLSEnabled    bool   `json:"ingressTLSEnabled"`
+	IngressTLSSecretName string `json:"ingressTLSSecretName"`
+
+	// External PostgreSQL
+	ExternalPG			  bool `json:ExternalPg`
+	ExternalPGHost        string `json:"ExternalPgHost"`
+	ExternalPGPort        int `json:"ExternalPgPort"`
+	ExternalPGUser        string `json:"ExternalPgUser"`
+	ExternalPGPassword    string `json:"ExternalPgUser"`
+	ExternalPGDataBase    string `json:"ExternalPgDataBase"`
+	ExternalPGSSLMode     string `json:"ExternalPgSSLMode"`
+	ExternalPGRootCASecret  string `json:"ExternalPgRootCASecret"`
+	ExternalPGClientSecret  string `json:"ExternalPgClientSecret"`
+
+	// Secrets
+	PGPassword             string `json:"pgPassword"`
+	PGExistingSecret       string `json:"pgExistingSecret"`
+}
+
+// NewHelmValuesFromCobraFlags returns an initialized HelmValuesFromCobraFlags
+func NewHelmValuesFromCobraFlags() *HelmValuesFromCobraFlags {
+	return &HelmValuesFromCobraFlags{
+		args:     map[string]interface{}{},
+		flagTree: FlagTree{},
+	}
+}
+
+// GetArgs returns the map of helm chart fields to values
+func (ctl *HelmValuesFromCobraFlags) GetArgs() map[string]interface{} {
+	return ctl.args
+}
+
```

Comment:
SetArgs method is missing which I have added recently to take care of upgrade scenario